Tests/MapboxDirectionsTests/IntersectionTests.swift

```diff
@@ -19,6 +19,10 @@ class IntersectionTests: XCTestCase {
                     "type": "toll_booth",
                     "name": "test toll booth"
                 ],
+                "railway_crossing": true,
+                "traffic_signal": true,
+                "stop_sign": false,
+                "yield_sign": false
```

Comment:
FYI, some of these combinations are possible, like a railroad crossing controlled by a traffic signal at an intersection. However, some combinations don’t make much sense, like both a stop sign and a yield sign. Unfortunately, a series of independent properties is probably the best we can do. An OptionSet would prevent some of these valid combinations and constrain us in the future if we need to add other forms of traffic control devices or other details about these devices, such as the kind of traffic signal (ramp meter, blink-mode, etc.).